core/src/main/java/org/bitcoinj/core/VersionMessage.java

```diff
@@ -98,7 +98,7 @@ public VersionMessage(NetworkParameters params, byte[] payload) throws ProtocolE
     public VersionMessage(NetworkParameters params, int newBestHeight) {
         super(params);
         clientVersion = params.getProtocolVersionNum(NetworkParameters.ProtocolVersion.CURRENT);
-        localServices = 0;
+        localServices = NODE_WITNESS;
```

Comment:
I haven't checked the corresponding code that announces newly-created transactions from the wallet to connected peers, but if these transactions are announced with an `inv` message including the transaction id, instead of a `tx` message, then the peer will follow up with a `getdata` message requesting the transaction. If the peer does not see that we are capable of providing witness data, it will request the transaction with an `type` field set to `MSG_TX`, which will be replied with the transaction serialised without witness data, consequently failing validation on a SegWit-capable node. The requesting node checks that we have `NODE_WITNESS` on our services field and requests `MSG_WITNESS_TX` when it wants to receive witness data for validation. This service flag is unrelated to being able to serve blocks or otherwise behaving as a full node.